decoder/vaapidecoder_h265.cpp

```diff
@@ -1088,6 +1096,52 @@ YamiStatus VaapiDecoderH265::decode(VideoDecodeBuffer* buffer)
     return YAMI_SUCCESS;
 }
 
+bool VaapiDecoderH265::decodeAvcRecordData(uint8_t* buf, int32_t bufSize)
+{
+    if (buf == NULL || bufSize == 0) {
+        ERROR("invalid record data");
+        return false;
+    }
+    if (buf[0] != 1) {
+        VideoDecodeBuffer buffer;
+        memset(&buffer, 0, sizeof(buffer));
+        buffer.data = buf;
+        buffer.size = bufSize;
+        return (decode(&buffer) == YAMI_SUCCESS);
+    }
+    if (bufSize < 24) {
+        ERROR("invalid avcc record data");
+        return false;
+    }
+    NalUnit nalu;
+    const uint8_t* nalBuf;
+    int32_t i = 0, j, numNalu, nalBufSize;
+    nalBuf = &buf[21];
+    m_nalLengthSize = (*nalBuf & 0x03) + 1;
```

Comment:
Don't init m_nalLengthSize if function decodeAvcRecordData() return false. Else common/NalReader.cpp will treat it as avcc format. Please refer  https://github.com/01org/libyami/blob/apache/decoder/vaapidecoder_h264.cpp#L1180